common/src/main/java/com/wynntils/utils/UnsignedByteUtils.java

```diff
@@ -69,71 +70,55 @@ public static String decodeString(List<UnsignedByte> bytes) {
     }
 
     public static UnsignedByte[] encodeVariableSizedInteger(long value) {
-        // Encode an signed integer value with a variable number of bytes.
-
-        // Do the encoding
-        UnsignedByte[] bytes;
-        if (value >= -128 && value <= 127) {
-            bytes = new UnsignedByte[1];
-            bytes[0] = UnsignedByte.of((byte) value);
-        } else if (value >= -32768 && value <= 32767) {
-            bytes = new UnsignedByte[2];
-            bytes[0] = UnsignedByte.of((byte) (value >> 8));
-            bytes[1] = UnsignedByte.of((byte) value);
-        } else if (value >= -8388608 && value <= 8388607) {
-            bytes = new UnsignedByte[3];
-            bytes[0] = UnsignedByte.of((byte) (value >> 16));
-            bytes[1] = UnsignedByte.of((byte) (value >> 8));
-            bytes[2] = UnsignedByte.of((byte) value);
-        } else if (value >= -2147483648 && value <= 2147483647) {
-            bytes = new UnsignedByte[4];
-            bytes[0] = UnsignedByte.of((byte) (value >> 24));
-            bytes[1] = UnsignedByte.of((byte) (value >> 16));
-            bytes[2] = UnsignedByte.of((byte) (value >> 8));
-            bytes[3] = UnsignedByte.of((byte) value);
-        } else {
-            bytes = new UnsignedByte[8];
-            bytes[0] = UnsignedByte.of((byte) (value >> 56));
-            bytes[1] = UnsignedByte.of((byte) (value >> 48));
-            bytes[2] = UnsignedByte.of((byte) (value >> 40));
-            bytes[3] = UnsignedByte.of((byte) (value >> 32));
-            bytes[4] = UnsignedByte.of((byte) (value >> 24));
-            bytes[5] = UnsignedByte.of((byte) (value >> 16));
-            bytes[6] = UnsignedByte.of((byte) (value >> 8));
-            bytes[7] = UnsignedByte.of((byte) value);
+        // Use zig-zag encoding to encode negative numbers
```

Comment:
TIL!

Reply:
The amount of black magic used here makes me a bit nervous, even though the method is tested throughly with many cases for longs, ints, signs, et..